rules/framework.bzl

```diff
@@ -340,20 +355,87 @@ def _apple_framework_packaging_impl(ctx):
             swift_common.create_module(name = swiftmodule_name, swift = swift_module),
         ]
 
-    # Eventually we need to remove any reference to objc provider
-    # and use CcInfo instead, see this issue for more details: https://github.com/bazelbuild/bazel/issues/10674
-    objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
+    direct_headers = []
+
+    if virtualize_frameworks:
```

Comment:
refactor wise: can we do if else and try to put them into sep. functions with clear inputs and outputs? Any reason not to do this? 🙏

Reply:
+1

Reply:
The key reason to keep the same code is so we don't make divergent code paths with this feature on/off - the code paths are very similar besides that one nooping the "clean" action: perhaps can rename a bit

Reply:
@congt  and @gyfelton I've taken the liberty to detangle some of the original code as we discussed offline, will also make it easier to remove the old way if we decide to do so!